gen/schema_gen_sum.go

```diff
@@ -836,6 +836,9 @@ func mergeSchemes(s1, s2 *jsonschema.Schema) (_ *jsonschema.Schema, err error) {
 		case s1.AdditionalProperties == nil && s2.AdditionalProperties != nil:
 			r.AdditionalProperties = s2.AdditionalProperties
 			r.Item = s2.Item
+		case reflect.DeepEqual(s1.AdditionalProperties, s2.AdditionalProperties) && reflect.DeepEqual(s1.Item, s2.Item):
```

Comment:
What if we try to merge `s1.Item` and `s2.Item` to `r.Item`? Would it work in your case?

Reply:
Yes, that would be fine. Shall I amend it to do so?